docs/howtos/setup-android-build-environment.md

```diff
@@ -108,10 +85,106 @@ you will find:
 
 You should also find the .kt files for the project somewhere there and in the right directory structure if that turns out to be useful.
 
+# Setting up the Android SDK and NDK.
+
+These instructions have recently worked. Older instructions that recently failed
+are below for reference.
+
+1. Install or locate the Android SDKs
+   - Install the Android SDKs. If Android Studio is involved, it may have already installed
+     them somewhere - use the "SDK Manager" to identify this location.
+   - Set `ANDROID_HOME` to this location and add it to your rc file.
+
+  If not already installed, you can install the "command-line" only tools:
+
+   - Visit https://developer.android.com/studio/, at the bottom of the page locate the current SDKs for linux
+     at time of writing, this is https://dl.google.com/android/repository/commandlinetools-linux-6514223_latest.zip
+
+  Install it by executing:
+
+  > % cd ~  
+  > % mkdir -p android-sdk/cmdline-tools  
+  > % cd android-sdk/cmdline-tools  
+  > % unzip {path-to.zip}  
+  > % export ANDROID_HOME=$HOME/android-sdk  
+  > % $ANDROID_HOME/cmdline-tools/tools/bin/sdkmanager "platforms;android-26"  
+  > % $ANDROID_HOME/cmdline-tools/tools/bin/sdkmanager --licenses  
+
+  Note that the The `cmdline-tools/tools` structure appears necessary for somewhat opaque reasons, see [stack overflow, obviously](https://stackoverflow.com/questions/60440509/android-command-line-tools-sdkmanager-always-shows-warning-could-not-create-se) for more.
+
+2. Install the Android NDK
+
+   - Install the NDK directly via the SDK manager. To work out exactly what
+     version you need, try executing `./gradlew assembleDebug`, for example, and
+     the build should fail telling to exactly what's missing - eg:
+
+  > Compatible side by side NDK version was not found. Default is 20.0.5594570.
+
+   - If you have the command-line tools:
+
+  > `$ANDROID_HOME/cmdline-tools/tools/bin/sdkmanager --install "ndk;20.0.5594570"
+
+   - If you have Android Studio or otherwise a GUI version of the SDK, use that
+     GUI.
+
+  In either case, you don't need to select where it is installed or to set
+  further environment variables.
+
+At the end of this process, all you need is `ANDROID_HOME` set!
+
+## Older instructions for the Android SDK and NDK.
+
+The more recent tools have simplified much of the process - however, at time of
+writing, this has only been tested on Ubuntu via WSL - so we're keeping this
+around until we get wider confirmation that it's out-dated.
+
+At the end of this process you should have the following environment variables set up.
+
+- `ANDROID_NDK_ROOT`
+- `ANDROID_NDK_HOME`
+- `ANDROID_NDK_API_VERSION`
+- `ANDROID_HOME`
+- `JAVA_HOME`
+
+These variables are required every time you build, so you should add them to
+a rc file or similar so they persist between reboots etc.
+
+1. Install the SDK, largely as described above.
+
+2. Install NDK r21 from https://developer.android.com/ndk/downloads
```

Comment:
I'm surprised that the instructions above seemed to install r20, but these outdated instructions call for r21. I guess it's related to the build error you mention below, but OTOH, I'm definitely using r21 and it seems to be working OK.